test/test_config.py

```diff
@@ -26,28 +20,24 @@ def mock_os_access() -> Generator[MagicMock, Any, None]:
         yield mock_access
 
 
-@pytest.fixture()
-def mock_os_path_abspath() -> Generator[MagicMock, Any, None]:
-    with patch("os.path.abspath") as mock_abspath:
-        yield mock_abspath
-
-
 def test_generate_config_fail_empty_config() -> None:
     with pytest.raises(RuntimeError, match="No config information found"):
         generate_config(None)
 
 
 def test_generate_config_with_valid_config(
-    mock_os_path_isabs: MagicMock,
-    mock_os_path_isdir: MagicMock,
+    monkeypatch: pytest.MonkeyPatch,
     mock_os_access: MagicMock,
-    mock_os_path_abspath: MagicMock,
 ) -> None:
-    mock_os_path_isabs.side_effect = lambda x: x.startswith("/")
-    mock_os_path_isdir.side_effect = lambda x: x.startswith("/")
     mock_os_access.side_effect = lambda x, y: True
-    mock_os_path_abspath.side_effect = lambda x: f"/absolute/{x}"
 
+    def starts_with_slash(self) -> bool:
+        """Fake function to replace is_absolute and is_dir."""
+        return str(self).startswith("/")
+
+    monkeypatch.setattr(Path, "is_absolute", starts_with_slash)
+    monkeypatch.setattr(Path, "is_dir", starts_with_slash)
+    monkeypatch.setattr(Path, "resolve", absolutify)
```

Comment:
replace the old mocks with these monkey patched versions (arguably a bit simpler?)